packages/rich-text/src/component/index.native.js

```diff
@@ -88,7 +88,7 @@ export class RichText extends Component {
 		this.formatToValue = memize( this.formatToValue.bind( this ), {
 			maxSize: 1,
 		} );
-
+		this.debounceCreateUndoLevel = debounce( this.onCreateUndoLevel, 1000 );
```

Comment:
I see here that we are debouncing this for every invocation, but web seems to have some cases where the `setTimeout` is not used, e.g. in `onChange` they call `this.onCreateUndoLevel` directly: https://github.com/WordPress/gutenberg/blob/33bace57b13aa307f8befb609d0798a6a16dbfaf/packages/rich-text/src/component/index.js#L669

I wonder, are there reasons for not debouncing on web in some cases? :thinking:

Reply:
We also have one case when we don't use debounce - onFormatChange which is also called in `onInsert`. I haven't changed the `onCreateUndoLevel` itself but i created a new function that is called `debounceCreateUndoLevel` and invoke it on each input change as they do here :) https://github.com/WordPress/gutenberg/blob/33bace57b13aa307f8befb609d0798a6a16dbfaf/packages/rich-text/src/component/index.js#L501

As far as i can see they also call `onChage` inside `onInput` but with `withoutHistory ` set to true which turns off creating of undo level. https://github.com/WordPress/gutenberg/blob/33bace57b13aa307f8befb609d0798a6a16dbfaf/packages/rich-text/src/component/index.js#L488

Reply:
Ah, right! Thanks for the response! I misread the constructor bind as a same-named function :man_facepalming: . Thanks for the explanation, and great work on improving this!